sunpy/map/nddata_compat.py

```diff
@@ -14,94 +14,81 @@
 
```

Comment:
i thought you have made a few extra conditional check on uncertainity and mask maybe